mesh/confirmation/views.py

```diff
@@ -0,0 +1,182 @@
+        verificationToken=verification_token
+    )
+
+    # Construct the confirmation URL
+    confirmation_url = request.build_absolute_uri(
+        f"/confirmation/{user_email}/{verification_token}/"
+    )
+
+    user_preferred_name = get_preferred_name(account_id)
+    
+    # Render the email template with the confirmation URL
+    email_subject = "Account Confirmation"
+    email_template = "confirmation_email.html"
+    email_context = {"confirmation_url": confirmation_url, "name": user_preferred_name}
+    email_message = render_to_string(email_template, email_context)
+    email_plain_message = strip_tags(email_message)
+
+    # Send the confirmation email
+    send_mail(
+        email_subject,
+        email_plain_message,
+        os.environ.get("EMAIL_NAME"),
+        [user_email],
+        html_message=email_message,
+    )
+
+    return HttpResponse(user_email + ": Confirmation email sent!")
+
+
+# Check that the user's stored token is the same one in the URL,
+# if so, set isVerified to true.
+def confirm_token(request, user_email, url_token):
+    # Grab account data by user_email
+    account_id = get_account_id(user_email)
+
+    # Grab settings data by the account_id
+    settings_data = get_settings_data(account_id)
+
+    email_already_verified = get_verification_status(account_id)
+
+    # Check if user email is verified yet or not
+    # if so, skip updating settings table
+    if email_already_verified:
+        return HttpResponse(
+            user_email
+            + ": Email confirmation success as user email is already verified!"
+        )
+
+    # Grab the token stored in the database for this user
+    db_token = settings_data["verificationToken"]
+
+    # If the tokens match, then the user is verified
+    if url_token == db_token:
+        Settings.objects.filter(accountID=account_id).update(isVerified=True)
+        return HttpResponse(user_email + ": Email confirmation success!")
+    else:
+        return HttpResponse(user_email + ": Email confirmation failed!")
+
+
+# Note: For the Django function Model.objects.filter(...), the function
+# returns a QuerySet, which is just a set of objects (python dicts).
+# If the user has a unique email/accountID, then the user should be the only item in the set,
+# and thus will be the at the 0th index. Each function checks if the user exists, if they don't
+# then the QuerySet will be empty.
+
+# Grab account ID by user_email
+def get_account_id(user_email):
+    raw_account_data = Account.objects.filter(email=user_email).values()
+    
+    if not raw_account_data:
+        return None
+
+    account_data = raw_account_data[0]
+    account_id = account_data["accountID"]
+
+    return account_id
+
+
+# Grab settings data by the account_id
+# settings_data is a python dict containing
+# data for each field in the Settings model
+def get_settings_data(account_id):
+    raw_settings_data = Settings.objects.filter(accountID=account_id).values()
```

Comment:
For efficiency purposes, it might be better to use `Settings.objects.get` since accountID is unique, meaning that there should be only one entry that matches the query.
[Query Get Documentation](https://docs.djangoproject.com/en/4.2/topics/db/queries/#retrieving-a-single-object-with-get)

Reply:
This is another good point that I will fix, as not only is it more efficient, the way I was doing it before required you to use the python dictionary subscripts to access parts of the object, whereas getting the object itself allows for autocomplete and generally more clear code.